coding.go

```diff
@@ -56,27 +55,21 @@ func fromImmutableNode(encoded *immutableProtoNode) *ProtoNode {
 		if next.FieldTsize().Exists() {
 			size = uint64(next.FieldTsize().Must().Int())
 		}
-		link := &format.Link{
-			Name: name,
-			Size: size,
-			Cid:  c,
-		}
-		links = append(links, link)
+		link := &linkAllocs[i]
+		link.Name = name
+		link.Size = size
+		link.Cid = c
+		n.links[i] = link
 	}
-	n.links = links
 	return n
 }
 func (n *ProtoNode) marshalImmutable() (*immutableProtoNode, error) {
 	nd, err := qp.BuildMap(dagpb.Type.PBNode, 2, func(ma ipld.MapAssembler) {
 		qp.MapEntry(ma, "Links", qp.List(int64(len(n.links)), func(la ipld.ListAssembler) {
 			for _, link := range n.links {
-				qp.ListEntry(la, qp.Map(-1, func(ma ipld.MapAssembler) {
+				qp.ListEntry(la, qp.Map(3, func(ma ipld.MapAssembler) {
```

Comment:
FWIW, I'm pretty sure the NodeBuilder for PBLink doesn't care about estimate cause it just makes a struct.

Reply:
I figured as much, and this indeed makes zero difference, but I thought it would be nice to do the right thing just in case.